File: 1). DSA/1). Data Structures/01). Array/C++/_037)_Max_Sum_of_K_Consecutive_Element.cpp
#include<iostream>
using namespace std;

int maxSumOf_K_Element(int arr[], int k, int n)
{
    int curr_sum = 0;
    for(int i=0; i<k; i++)
        curr_sum += arr[i];
    int max_sum = curr_sum;
    for(int i=k; i<n; i++)
    {
        curr_sum += (arr[i] - arr[i-k]);
        max_sum = max(max_sum, curr_sum);
    }
    return max_sum;
}

int main()
{
    int arr[] = {1, 8, 30, -5, 20, 7}, k=3, n=6;
    cout<<"Max Sum of 3 Consecutive Element = "<<maxSumOf_K_Element(arr, k, n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_037)_Max_Sum_of_K_Consecutive_Element.dart
import 'dart:math';

int maxSumOfKElements(List<int> arr, int k) {
  int currMax = 0;
  for (int i = 0; i < k; i++) {
    currMax += arr[i];
  }
  int maxSum = currMax;

  for (int i = k; i < arr.length; i++) {
    currMax += (arr[i] - arr[i - k]);
    maxSum = max(maxSum, currMax);
  }
  return maxSum;
}

void main() {
  var arr = [1, 8, 30, -5, 20, 7];
  int k = 3;
  print(
      'Maximum Sum of $k consecutive elements = ${maxSumOfKElements(arr, k)}');
}
